File: platform-sdk-java/platform-sdk-task/src/main/java/com/platform/sdk/task/TaskCompletionBuilder.java
package com.platform.sdk.task;

import com.fasterxml.jackson.core.type.TypeReference;
import com.platform.sdk.core.http.PlatformHttpClient;

import java.io.IOException;
import java.util.HashMap;
import java.util.Map;

public class TaskCompletionBuilder {

    private final PlatformHttpClient http;
    private final String taskId;
    private String outcome;
    private final Map<String, Object> variables = new HashMap<>();

    TaskCompletionBuilder(PlatformHttpClient http, String taskId) {
        this.http = http;
        this.taskId = taskId;
    }

    public TaskCompletionBuilder outcome(String outcome) {
        this.outcome = outcome;
        return this;
    }

    public TaskCompletionBuilder variable(String name, Object value) {
        this.variables.put(name, value);
        return this;
    }

    public TaskCompletionBuilder formData(Map<String, Object> data) {
        this.variables.putAll(data);
        return this;
    }

    public void submit() throws IOException {
        Map<String, Object> body = new HashMap<>();
        if (outcome != null) body.put("outcome", outcome);
        if (!variables.isEmpty()) body.put("variables", variables);
        http.post("/v1/tasks/" + taskId + "/complete", body, new TypeReference<Void>() {});
    }
}

File: platform-sdk-js/src/task/TaskCompletionBuilder.ts
import { PlatformClient } from '../client/PlatformClient'

export class TaskCompletionBuilder {
  private _outcome?: string
  private readonly _variables: Record<string, unknown> = {}

  constructor(private readonly client: PlatformClient, private readonly taskId: string) {}

  outcome(outcome: string): this { this._outcome = outcome; return this }
  variable(name: string, value: unknown): this { this._variables[name] = value; return this }
  formData(data: Record<string, unknown>): this { Object.assign(this._variables, data); return this }

  async submit(): Promise<void> {
    const body: Record<string, unknown> = {}
    if (this._outcome) body.outcome = this._outcome
    if (Object.keys(this._variables).length > 0) body.variables = this._variables
    return this.client.post(`/v1/tasks/${this.taskId}/complete`, body)
  }
}
